### ISSUE TITLE
IndexError: index 5 is out of bounds for axis 0 with size 5

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#3086]
to reproduce: 

Data: [Archive.zip](https://github.com/neuropoly/spinalcordtoolbox/files/5656387/Archive.zip)

<details><summary>syntax and error</summary>

```
sct_qc -i mt0.nii.gz -s label.nii -p sct_label_utils -qc qc

--
Spinal Cord Toolbox (git-jca/3044-ants-sampling-770e37f8b25f3283ede65eb171501f55bf5416e5)

sct_qc -i mt0.nii.gz -s label.nii -p sct_label_utils -qc qc
--


*** Generate Quality Control (QC) html report ***
Resample images to NonexNone mm
QcImage: layout with Sagittal slice
Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_qc.py", line 87, in <module>
    main(arguments)
  File "/Users/julien/code/sct/scripts/sct_qc.py", line 81, in main
    process=args.p)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 712, in generate_qc
    angle_line=angle_line
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 571, in add_entry
    layout(qcslice)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 274, in wrapped_f
    img, mask = func(sct_slice, *args)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 569, in layout
    return qcslice_layout(qslice)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 678, in qcslice_layout
    def qcslice_layout(x): return x.single()
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/slice.py", line 291, in single
    matrix[row] = self.get_slice(image.data, int(np.round(index[row])))[row]
IndexError: index 5 is out of bounds for axis 0 with size 5
```

</details>

**Comment by joshuacwnewton** [comment]
Hrm, I was unable to reproduce this error:

<details>
<summary> Console output </summary>

```console
> sct_qc -i mt0.nii.gz -s label.nii -p sct_label_utils -qc qc

--
Spinal Cord Toolbox (git-jn/3086-fix-qc-index-error-c7810852dc0908e5a1bdede2e544eb8592817b7c)

sct_qc -i mt0.nii.gz -s label.nii -p sct_label_utils -qc qc
--


*** Generate Quality Control (QC) html report ***
Resample images to NonexNone mm
QcImage: layout with Sagittal slice
/home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/lib/python3.6/site-packages/matplotlib/image.py:446: UserWarning: Warning: converting a masked element to nan.
  dv = np.float64(self.norm.vmax) - np.float64(self.norm.vmin)
/home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/lib/python3.6/site-packages/matplotlib/image.py:453: UserWarning: Warning: converting a masked element to nan.
  a_min = np.float64(newmin)
/home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/lib/python3.6/site-packages/matplotlib/image.py:458: UserWarning: Warning: converting a masked element to nan.
  a_max = np.float64(newmax)
Successfully generated the QC results in qc/_json/qc_2021_02_18_115240.538486.json
Use the following command to see the results in a browser:
xdg-open "qc/index.html"

```
</details>

![Screenshot from 2021-02-18 11-50-59](https://user-images.githubusercontent.com/16181459/108391341-9c5e3f00-71df-11eb-8d6e-ffb472bc957e.png)

Is this the expected output?

**Comment by jcohenadad** [comment]
> Is this the expected output?

Yes it is. I tried again with a recent commit (c7810852dc0908e5a1bdede2e544eb8592817b7c) and I am unable to reproduce the error either 🤷‍♂️ . 

Oh well, that was an easy fix 😊
